### PR TITLE
(feat) add user initials icon component

### PR DESCRIPTION
**Pull request by benjaminstrasser**
This PR introduces a new component to generate a user initials avatar/icon with a dynamically generated background color based on the user's first and last name.

### CODE REVIEW

File: src/components/ui/user-icon/user-default-icon.svelte [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<script lang="ts">
+	export const firstname: string = 'Nomen'
+	export const lastname: string = 'Nescio'
+
+	function hashCode(str: string) {
+		let hash = 0
+		for (let i = 0, len = str.length; i < len; i++) {
+			let chr = str.charCodeAt(i)
+			hash = (hash << 3) - hash + chr
+			hash |= 0
```
Comment (mledl) [inline src/components/ui/user-icon/user-default-icon.svelte:10]: I think bitwise-or with 0 does not change the value of `hash` except the calculation above leads to a negative number which is represented in Two's complement.
Is that the case?

File: src/components/ui/user-icon/user-default-icon.svelte [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<script lang="ts">
+	export const firstname: string = 'Nomen'
+	export const lastname: string = 'Nescio'
+
+	function hashCode(str: string) {
```
Comment (mledl) [inline src/components/ui/user-icon/user-default-icon.svelte:5]: Can we either write some component test for this logic or put it into some util and test it there?

File: src/components/ui/user-icon/user-default-icon.svelte (hunk 1 above)
Comment (benjaminstrasser) [reply src/components/ui/user-icon/user-default-icon.svelte:10]: Yeah this is just a common js quirk, where it converts the number(IEEE 754 double precision) into a 32bit number in two's compliment.

File: src/components/ui/user-icon/user-default-icon.svelte (hunk 2 above)
Comment (benjaminstrasser) [reply src/components/ui/user-icon/user-default-icon.svelte:5]: imo once we start testing this for uniformity and collisions, we should just use crypto-js or polyfill node:crypto. Then we could just use md5.
What is your opinion?

Comment (mledl) [reply src/components/ui/user-icon/user-default-icon.svelte:5]: I think this is good enough for now.

**Review by mledl — APPROVED** [review]

**Pull request merged**